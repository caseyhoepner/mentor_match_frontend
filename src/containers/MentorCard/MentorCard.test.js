import React from 'react';
import { shallow } from 'enzyme';
import { MentorCard } from './';

describe('MentorCard', () => {
  let wrapper;
  let mockMentor = {
    props: {
      mentor: {
        name: 'Oscars Blues',
        current_title: 'Coder',
        background: 'I know how to code!'
      }
    }
  }
  
  beforeEach(() => {
    wrapper = shallow(<MentorCard mentor={mockMentor} />)
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});