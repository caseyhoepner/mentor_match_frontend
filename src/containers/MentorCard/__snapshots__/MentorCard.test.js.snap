// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MentorCard matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MentorCard
    mentor={
      Object {
        "props": Object {
          "mentor": Object {
            "background": "I know how to code!",
            "current_title": "Coder",
            "name": "Oscars Blues",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Default profile"
          className="mcard-pic"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "mentor-pic-default.svg",
            }
          }
        />,
        <h2>
          Oscars Blues
        </h2>,
        <p>
          Coder
        </p>,
        <p>
          I know how to code!
        </p>,
      ],
      "className": "mcard-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Default profile",
          "className": "mcard-pic",
          "src": Object {
            "ReactComponent": [Function],
            "__esModule": true,
            "default": "mentor-pic-default.svg",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Oscars Blues",
        },
        "ref": null,
        "rendered": "Oscars Blues",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Coder",
        },
        "ref": null,
        "rendered": "Coder",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "I know how to code!",
        },
        "ref": null,
        "rendered": "I know how to code!",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Default profile"
            className="mcard-pic"
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "mentor-pic-default.svg",
              }
            }
          />,
          <h2>
            Oscars Blues
          </h2>,
          <p>
            Coder
          </p>,
          <p>
            I know how to code!
          </p>,
        ],
        "className": "mcard-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Default profile",
            "className": "mcard-pic",
            "src": Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "mentor-pic-default.svg",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Oscars Blues",
          },
          "ref": null,
          "rendered": "Oscars Blues",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Coder",
          },
          "ref": null,
          "rendered": "Coder",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "I know how to code!",
          },
          "ref": null,
          "rendered": "I know how to code!",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
